### PR TITLE
apps: use external config when decoding from rc annotation

### PR DESCRIPTION
**Pull request by mfojtik**
This move strategies, hooks and deployer to use external deployment config when decoding it from the replication controller annotation. That means the `pkg/apps/strategy` and the deployer command are now using the external api exclusively.

The last users of internal DC is storage (REST) and controllers which should be refactored as follow up.

/cc @tnozicka 
/cc @deads2k

### CODE REVIEW

File: pkg/apps/util/const.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package util
+
+const (
```
Comment (mfojtik) [inline pkg/apps/util/const.go]: @deads2k @tnozicka i tried to organize this a little... if the const is being used only by the utils itself, make it private... I will try to get rid of `DeploymentStatusAnnotation`.

File: pkg/apps/strategy/support/lifecycle.go [hunk 2]
```diff
@@ -18,25 +18,30 @@ import (
...
 	"github.com/openshift/origin/pkg/util"
 )
 
-// hookContainerName is the name used for the container that runs inside hook pods.
-const hookContainerName = "lifecycle"
+const (
+	// hookContainerName is the name used for the container that runs inside hook pods.
+	hookContainerName = "lifecycle"
+	// deploymentPodTypeLabel is a label with which contains a type of deployment pod.
+	deploymentPodTypeLabel = "openshift.io/deployer-pod.type"
```
Comment (mfojtik) [inline pkg/apps/strategy/support/lifecycle.go]: @tnozicka move to the user, we don't have to expose this as API afaik.

File: pkg/apps/util/scheme_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const legacyDC = `{
+  "apiVersion": "v1",
+  "kind": "DeploymentConfig",
+  "metadata": {
+    "name": "sinatra-app-example-a"
+  }
+}
+`
+
+func TestLegacyDecoding(t *testing.T) {
```
Comment (mfojtik) [inline pkg/apps/util/scheme_test.go]: @deads2k re-added your test, just changed the codec to decode to external

**Comment by deads2k** [comment]
Assuming values are read from the server or serialized from the server, they will already be defaulted.

/lgtm

**Comment by tnozicka** [comment]
the two top commits lgtm
should this wait for https://github.com/openshift/origin/pull/20341 to merge or that pull should be canceled and go all in one here? (Putting hold temporarily, feel free to remove it when you choose the path.)
/hold

**Comment by mfojtik** [comment]
lets merge all here, tested together.

**Pull request merged**